genesis/engine/solvers/rigid/gjk_decomp.py

```diff
@@ -150,35 +187,40 @@ def __init__(self, rigid_solver):
         # because a face has 3 edges.
         self.polytope_horizon_stack = struct_polytope_horizon_data.field(shape=(self._B, self.polytope_max_faces * 3))
 
-        ### Multi-contact
-        # Whether or not to use MuJoCo's multi-contact detection algorithm. Only when the MuJoCo compatibility is
-        # enabled and the multi-contact detection is enabled, we use the algorithm.
-        self.enable_mujoco_multi_contact = (
-            self._solver._enable_multi_contact and self._solver._enable_mujoco_compatibility
-        )
+        # Threshold for reprojection error when we compute the witness points from the polytope. In computing the
+        # witness points, we project the origin onto the polytope faces and compute the barycentric coordinates of the
+        # projected point. To confirm the projection is valid, we compute the projected point using the barycentric
+        # coordinates and compare it with the original projected point. If the difference is larger than this threshold,
+        # we consider the projection invalid, because it means numerical errors are too large.
+        self.polytope_max_reprojection_error = gs.np_float(1e-5)
```

Comment:
How did you set this threshold? Are you confident about this value? Has it been extensively tested? Would be nice to add one extra comment line to explain in what extent this value should be blindly trusted or can be further tuned.